weave/trace/weave_client.py

```diff
@@ -799,19 +799,23 @@ def _save_nested_objects(self, obj: Any, name: Optional[str] = None) -> Any:
             if ref.project == self.project:
                 return
             remove_ref(obj)
+        # Must defer import here to avoid circular import
+        from weave.flow.obj import Object
 
-        if isinstance(obj, (pydantic.BaseModel, pydantic.v1.BaseModel)):
+        if isinstance(obj, Object):
             obj_rec = pydantic_object_record(obj)
             for v in obj_rec.__dict__.values():
                 self._save_nested_objects(v)
             ref = self._save_object_basic(obj_rec, name or get_obj_name(obj_rec))
             obj.__dict__["ref"] = ref
+        elif isinstance(obj, (pydantic.BaseModel, pydantic.v1.BaseModel)):
+            obj_rec = pydantic_object_record(obj)
```

Comment:
Let's sync on this - idk if `pydantic_object_record` is really the right dump here. We get those extra types/base classes/etc which seem to only make sense if the object is deserializable.

Reply:
That is the correct thing to do. It should be the same thing as what we actually save objects as.

Reply:
hmmm interesting. Ok